anomalib/post_processing/visualizer.py

```diff
@@ -161,6 +161,7 @@ def _visualize_full(self, image_result: ImageResult) -> np.ndarray:
             visualization.add_image(image=image_result.segmentations, title="Segmentation Result")
         elif self.task == TaskType.CLASSIFICATION:
             visualization.add_image(image_result.image, title="Image")
+            visualization.add_image(image_result.heat_map, "Predicted Heat Map")
```

Comment:
Is this intentional? If so, can you explain this change?

Reply:
My intention here is that even in a classification task, the model might predict a heatmap that can be useful to the user.

Reply:
I see. I agree it would be useful to show the heatmap for classification models, but not all classification models produce a heatmap, so this might break. So we need to add a check here to make sure that the heatmap is available:
```suggestion
            if hasattr(image_result, "heat_map"):
                visualization.add_image(image_result.heat_map, "Predicted Heat Map")
```